Lib/weakref.py

```diff
@@ -384,13 +391,18 @@ def _commit_removals(self):
         # because a dead weakref never compares equal to a live weakref,
         # even if they happened to refer to equal objects.
         # However, it means keys may already have been removed.
-        l = self._pending_removals
+        pop = self._pending_removals.pop
         d = self.data
-        while l:
+        while True:
             try:
-                del d[l.pop()]
-            except KeyError:
-                pass
+                key = pop()
+            except IndexError:
+                return
+            else:
+                try:
+                    del d[key]
+                except KeyError:
+                    pass
```

Comment:
Nesting here is unnecessary since the `except` block returns.
```suggestion

            try:
                del d[key]
            except KeyError:
                pass
```